azurerm/helpers/validate/database.go

```diff
@@ -12,7 +12,7 @@ func DatabaseCollation(i interface{}, k string) (warnings []string, errors []err
 		return
 	}
 
-	matched, _ := regexp.MatchString(`^[A-Za-z0-9_. ]+$`, v)
+	matched, _ := regexp.MatchString(`^[-A-Za-z0-9_. ]+$`, v)
 
 	if !matched {
 		errors = append(errors, fmt.Errorf("%s contains invalid characters, only underscores are supported, got %s", k, v))
```

Comment:
could we update this message?

Reply:
Hi @katbyte,
thanks for your comment, error message is updated in new commit.